### PR TITLE
Added support for naming a task step

### PR DESCRIPTION
**Pull request by dbeuchler**
Added support to set the name of a task step

This should be merged for 5.3

### CODE REVIEW

File: src/Moryx.AbstractionLayer/Activities/Activity.cs [hunk 1]
```diff
@@ -48,6 +54,9 @@ public abstract class Activity : IActivity
         internal Activity()
         {
             Tracing = new Tracing();
+
+            var displayName = GetType().GetDisplayName();
+            Name = !string.IsNullOrEmpty(displayName) ? displayName : GetType().Name;
```
Comment (Toxantron) [inline src/Moryx.AbstractionLayer/Activities/Activity.cs:59]: We should invert this. The `!` is easily overlooked and requires an unnecessary CPU cycle.

File: src/Moryx.AbstractionLayer/Tasks/TaskStep.cs [hunk 2]
```diff
@@ -32,13 +33,22 @@ public abstract class TaskStep<TActivity, TProcParam, TParam> : WorkplanStepBase
...
+
         /// <summary>
         /// Instantiate task and provide possible results
         /// </summary>
         protected TaskStep()
         {
             Parameters = new TParam();
 
+            var displayName = GetType().GetDisplayName();
+            ((INamedTaskStep)this).Name = !string.IsNullOrEmpty(displayName) ? displayName : GetType().Name;
```
Comment (Toxantron) [inline src/Moryx.AbstractionLayer/Tasks/TaskStep.cs:50]: Same

File: src/Moryx.Products.Management/Implementation/RecipeStorage.cs [hunk 3]
```diff
@@ -133,18 +133,16 @@ private static List<IWorkplanStep> LoadSteps(WorkplanEntity workplan, IDictionar
...
                 // Restore Subworkplan if necessary
-                var subworkplanStep = step as ISubworkplanStep;
-                if (subworkplanStep != null)
-                {
+                if (step is ISubworkplanStep subworkplanStep)
                     subworkplanStep.Workplan = LoadWorkplan(stepEntity.SubWorkplan);
-                }
+
+                // Restore step name
+                if (step is INamedTaskStep namedTaskStep)
```
Comment (Toxantron) [inline src/Moryx.Products.Management/Implementation/RecipeStorage.cs:144]: Maybe extend the condition if the entity value is not `NullOrEmpty`. In that case we should keep the constructor initialized Name

**Review by Toxantron — CHANGES_REQUESTED** [review]

File: src/Moryx.AbstractionLayer/Activities/Activity.cs (hunk 1 above)
Comment (dbeuchler) [reply src/Moryx.AbstractionLayer/Activities/Activity.cs]: done

File: src/Moryx.AbstractionLayer/Tasks/TaskStep.cs (hunk 2 above)
Comment (dbeuchler) [reply src/Moryx.AbstractionLayer/Tasks/TaskStep.cs]: done

File: src/Moryx.Products.Management/Implementation/RecipeStorage.cs (hunk 3 above)
Comment (dbeuchler) [reply src/Moryx.Products.Management/Implementation/RecipeStorage.cs]: done

**Review by Toxantron — APPROVED** [review]

**Pull request merged**